Fix legacy html volpage-loading

chansikpark opened the pull request:
This resolves #1518 and replaces PR #1516. Details in commit comments.

Resultant volpages were manually tested within Arango against what they were before the first change to `pali_concord.json` by running the old algorithm against the old data and subsequently re-running the new algorithm against the new data.

Aside from changes resulting from (I assume) debaking (ie. in Jatakas, Sutta Nipata), the Mahasangiti references for the Cariyapitaka (eg "ms21C1" vs "ms21Cp_2") were different in a way that I wasn't sure were intentional or not. These references no longer show up in suttaplex.

Also, some of the pts-vp-pli references for Bv seem to have typos (eg. "pts-vp-pli1ed.17" vs "pts-vp-pli1ed1.17"). These show up in suttaplex without the volume number.

***

I should note that in the process of deprecating the previous PR #1516 and wanting to fold the development on this issue neatly into a topic branch, I've complicated the commit history unnecessarily.

This branch's ref was mistakenly pushed while its head was still pointing to the starting point of my master branch. Instead, it should have been rebased to point to the HEAD of the master branch before pushing. This has led to an unnecessary complication of what was already an unnecessarily complicated commit history.

I attach an image here that may or may not clarify the history. The commit nodes are labelled 1 through 9 to correspond to the 9 commits below.

[image removed by Aminah]

Note that this commit's parents are the eigth and third commits.

It should all just work, but auditing the history may be a bit taxing. If need be, I could create another account or something to maintain a cleaner fork.

sujato commented:
Thanks so much for this, it is *extremely* valuable. I would really appreciate it if you would look into these references as much as possible to ensure we resolve any outstanding issues.

With the Bv issue, I have fixed it.

With Cp, I believe I have fixed it, but it is not yet perfect. It seems the underlying issue here is that the source files registered the first segment in a series as either 1.1 or 1.1.1, which then appear as separate segments, whereas obviously they are the same. I have used a simple regex to resolve this, standardizing the ID forms as I do so. However, there are certainly still some duplicate references. It is expected that, since we are merging data from different sources, we will have off-by-one errors. Any deduping of these would be helpful. As a rule, if a segment ID is repeated on a subsequent segment, delete the second one.

chansikpark commented:
Glad to hear it, sir.

> I would really appreciate it if you would look into these references as much as possible to ensure we resolve any outstanding issues.

Unless advised otherwise, I'll start looking into the testing framework that's in place and see if there's a nominal way to maintain tests for data integrity with reference to all relevant issues. 

...and after the stop-gap on #1353, I'll also look at what the code might be like if it were integrated with bilara-data and the dependencies on pootle files were removed.

> With the Bv issue, I have fixed it.

Verified on my machine.

> With Cp, I believe I have fixed it, but it is not yet perfect.

I'll submit a PR in fork.

chansikpark commented:
Obsoleted by #1537
